File: WjAnBase/src/main/java/com/abase/okhttp/Interceptor/DownInterceptor.java
package com.abase.okhttp.Interceptor;

import com.abase.okhttp.OhFileCallBakListener;
import com.abase.okhttp.ProgressResponseBody;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Response;

/**
 * 下载进度拦截
 * @author Admin
 * @version 1.0
 * @date 2018/1/5
 */

public class DownInterceptor implements Interceptor {
    private OhFileCallBakListener callbackListener;

    public DownInterceptor(OhFileCallBakListener callbackListener) {
        this.callbackListener = callbackListener;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        //拦截
        Response originalResponse = chain.proceed(chain.request());
        //包装响应体并返回
        return originalResponse.newBuilder().body(new
                ProgressResponseBody(originalResponse.body(),
                callbackListener)).build();
    }
}

File: ktlibrary/src/main/java/com/wj/okhttp/Interceptor/DownInterceptor.kt
package com.wj.okhttp.Interceptor

import java.io.IOException
import okhttp3.Response
import okhttp3.Interceptor
import kotlin.Throws
import com.wj.okhttp.*

/**
 * 下载进度拦截
 * @author Admin
 * @version 1.0
 * @date 2018/1/5
 */
class DownInterceptor(private val callbackListener: OhFileCallBakListener) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        //拦截
        val originalResponse: Response = chain.proceed(chain.request())
        //包装响应体并返回
        return originalResponse.newBuilder().body(ProgressResponseBody(originalResponse.body,
            callbackListener)).build()
    }
}